Downloaded content changes

deeppandya opened the pull request.

vinodghai committed:
Playlist UI XML is created

BottomSheet UI Fixes

Exoplayer Added.

deeppandya committed:
Add changes to work with playlist button

Finish flow from the web page to player

deeppandya committed:
Update code with viewmodel

Update interaction

Remove activities and create fragments instead

Update existing code to work with rename functionality

deeppandya committed:
Change package name

deeppandya committed:
Update player code to address queue

Update option icons

Add option to delete playlist

deeppandya committed:
Add all playlist fragment

deeppandya committed:
Add notification for the player

Address notification changes for android12,13

deeppandya committed:
Add bottom view for playlist player

deeppandya committed:
Add full screen layout

deeppandya committed:
Add cast feature

Add cast with platlist

Add media session deps

Update code for all playlists

deeppandya committed:
Update menu interaction

Add check for expire content

Add playlist menu onboarding

Update playlist options enums

deeppandya committed:
Update playlist button onboarding

Set default plalist update

Update menu interactions

Resolve full screen UI

Rename playlist option

Online/Offline mode update

deeppandya committed:
Add playlist events

Update move/copy functions

Update behavioural changes

deeppandya committed:
Add connection states

Update default playlist behaviour

deeppandya committed:
Add new playlist in move/copy section

Update thumbnail images

Update onboarding text

Update strings based on feedback

Update utils

Format code

deeppandya committed:
Resolve full screen view for ythe player

Resolve filesize

Refactor code

Rempove json dependency

Refactor code

deeppandya committed:
Add support for vector drawable support

deeppandya committed:
ADd last saved position

Add local db

update few behavioural changes

deeppandya committed:
Update playlist button onboarding

Update onboarding code

Resolve playback service issue

Update tablet UI

Resolve double playback issue

Remove currently playes song

Update changes for playlist service

Update player notification adapter

Update playlist icon tint

deeppandya committed:
Add implementation for notification click

deeppandya committed:
Update player notification click event

Update icon tint for light mode

Add tint to current playing item

Update onboarding

Add exoplayer hls dependency

Add HLS support

deeppandya committed:
Make changes for events

deeppandya committed:
Add brave licenses

Update minor ui changes

Update gradle file for aar signing

Update private variables

Remove commented code

deeppandya committed:
Add shell script to build .aar

deeppandya committed:
Create README.md

deeppandya committed:
Update code to sign and verify .aar details README.md

Apply changes to hide playlist button

Update tint color for hide icon

deeppandya committed:
Improve full screen UI with better ratio

Address security concerns

Add newline

Address PR comments

deeppandya committed:
Remove broadcast and work with livedata to expose service data

Add test HLS to assets

Update warnings

deeppandya committed:
Initial work for partial downloaded file

deeppandya committed:
Add hlsMediaPath field for android

Update hlsMediaPath after download

Update existing changes with hls file

deeppandya committed:
Add changes to download multiple HLS items

deeppandya committed:
Update recycler view adapetrs with list adapter

Resolve rebase issue with requireActivity on back press

deeppandya committed:
Update playlist item events

deeppandya committed:
Initial commit for exoplayer migration

Resolve media session issue

deeppandya committed:
Update UI changes for download state

Reformat code

Remove offline option form playlist item swipe

deeppandya committed:
Update play icons

Update onboarding rounded bg

deeppandya committed:
Remove unused resources from kotlin module

deeppandya committed:
Refactor HLS content option

Refactor and remove unused commented changes

deeppandya commented:
Added @stoletheminerals for security/privacy concerns

deeppandya commented on src/main/java/com/brave/playlist/playback_service/VideoPlaybackService.kt:
```diff
@@ -0,0 +1,223 @@
+    }
+
+    private fun initializeSessionAndPlayer() {
+        val dataSourceFactory = DataSource.Factory { FileDataSource.Factory().createDataSource() }
```
Explicitly mentioned `FileDataSource` as DataSource which only works for File Uri

deeppandya committed:
Update add media icon

Update player bottom sheet behaviour

samartnik reviewed (approved):
lgtm

SergeyZhukovsky commented:
@deeppandya does that PR have anything new or what the actual changes? As from what I see it's basically some kind of a rebase of old commits. Do we need to review everything or just certain commits?

deeppandya commented:
> @deeppandya does that PR have anything new or what the actual changes? As from what I see it's basically some kind of a rebase of old commits. Do we need to review everything or just certain commits?

@SergeyZhukovsky I have changed the implementation to work with his content differently. This kotlin module is mainly for the UI and player interaction. I would be better if you can briefly go over the changes as I have changed almost every files with new changes. I am sorry for the trouble but there is no easy way for it.

SergeyZhukovsky reviewed (approved):
++

deeppandya committed:
Resolve db migration

Add schema location

deeppandya committed:
Remove swipe to delete for non cached item

tapanmodh commented on src/main/java/com/brave/playlist/fragment/PlaylistFragment.kt:
```diff
@@ -0,0 +1,596 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
```
Two times copyright

tapanmodh commented on .gitignore:
```diff
@@ -0,0 +1,27 @@
+
+# IntelliJ
+*.iml
+.idea/
```
nit : newline at the end

tapanmodh commented on schemas/com.brave.playlist.local_database.PlaylistDatabase/1.json:
```diff
@@ -0,0 +1,66 @@
+      "INSERT OR REPLACE INTO room_master_table (id,identity_hash) VALUES(42, '97917132ab7ccac87386de3a7faebfce')"
+    ]
+  }
+}
```
nit : newline at the end

deeppandya replied:
it's an autogenerated file so we don't need to have newline at the end.

tapanmodh commented on src/main/res/drawable/ic_button_bg.xml:
```diff
@@ -0,0 +1,12 @@
+    android:shape="rectangle">
+    <corners android:radius="48dp" />
+    <solid android:color="#4C54D2" />
+</shape>
```
nit : newline at the end

tapanmodh commented on src/main/res/drawable/ic_button_unfilled_bg.xml:
```diff
@@ -0,0 +1,14 @@
+    <stroke
+        android:width="1dp"
+        android:color="#5E6175" />
+</shape>
```
nit : newline at the end

deeppandya committed:
Add changes for delet/reorder playlist items

deeppandya committed:
Allow delete while in progress

Changes related to delete while playing

tapanmodh reviewed (approved):
lgtm
